internal/api/list_stages_v1alpha1.go

```diff
@@ -29,6 +30,10 @@ func (s *server) ListStages(
 		return nil, fmt.Errorf("list stages: %w", err)
 	}
 
+	sort.Slice(list.Items, func(i, j int) bool {
+		return list.Items[i].Name < list.Items[j].Name
+	})
```

Comment:
Since Go 1.21, this is the preferred way to sort.

```suggestion
	slices.SortFunc(list.Items, func(a, b kargoapi.Stage) int {
		return strings.Compare(a.Name, b.Name)
	})
```

Note, we do have a lot of places in the code base where it's still done the older way and I'll open a separate chore issue for fixing those.

Reply:
Ah thanks for catching that! Updated the ListProject sort too. Sorry for the back and forth